### PR TITLE
Minor class definition updates for AWSGem

### PR DESCRIPTION
**Pull request by spham-amzn**
Move the queued-events constructors for AWSApiClientJob and ServiceRequestJob from public to protected, and add short description for them

Signed-off-by: Steve Pham <82231385+spham-amzn@users.noreply.github.com>

### CODE REVIEW

File: Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h [hunk 1]
```diff
@@ -232,6 +210,32 @@ namespace AWSCore
...
+                         bool queueOnFailure,
+                         OnFailureFunction onFailure,
+                         bool queueDelete,
+                         IConfig* config = GetDefaultConfig()
+        ) : AwsApiClientJobType(false, config)
+          , m_queueOnSuccess{ queueOnSuccess }
+          , m_onSuccess{ onSuccess }
+          , m_queueOnFailure{ queueOnFailure }
+          , m_onFailure{ onFailure }
+          , m_queueDelete{ queueDelete }
```
Comment (onecent1101) [inline Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h:226]: Minor: actually you don't need this constructor? just move this part to 
```
AwsApiRequestJob(OnSuccessFunction onSuccess, OnFailureFunction onFailure, IConfig* config = GetDefaultConfig())
```
same for another class

**Review by onecent1101 — APPROVED** [review]

File: Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h [hunk 2]
```diff
@@ -232,6 +210,32 @@ namespace AWSCore
         }
 
     protected:
+        /// Constructor for creating Jobs that can handle queued responses
```
Comment (junbo75) [inline Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h:213]: Nit: May worth specifying the job type (AwsApiRequestJob) in the description.

File: Gems/AWSCore/Code/Include/Framework/ServiceRequestJob.h [hunk 3]
```diff
@@ -241,6 +218,33 @@ namespace AWSCore
         /// for replacing these parts of the url.
         const Aws::String& m_requestUrl;
 
+        /// Constructor for creating Jobs that can handle queued responses
```
Comment (junbo75) [inline Gems/AWSCore/Code/Include/Framework/ServiceRequestJob.h:221]: Nit: May worth specifying the job type (ServiceRequestJob) in the description.

**Review by junbo75 — APPROVED** [review]

File: Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h (hunk 1 above)
Comment (spham-amzn) [reply Gems/AWSCore/Code/Include/Framework/AWSApiRequestJob.h:226]: Probably not. Will consolidate them

File: Gems/AWSCore/Code/Include/Framework/ServiceRequestJob.h (hunk 3 above)
Comment (spham-amzn) [reply Gems/AWSCore/Code/Include/Framework/ServiceRequestJob.h]: done

**Pull request merged**